src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -146,6 +146,29 @@ public function onFinishRequest(FinishRequestEvent $event)
         }
     }
 
+    public function onSessionUsage(): void
+    {
+        if (!$this->debug) {
+            return;
+        }
+
+        if (!$requestStack = $this->container && $this->container->has('request_stack') ? $this->container->get('request_stack') : null) {
+            return;
+        }
+
+        if ($requestStack->getMasterRequest()->attributes->get('_stateless')) {
```

Comment:
I think we should look at the full stack instead of just the master request.

Reply:
Yes you're right !

But what happens if a request in the stack is declared as stateless and another request in the stack uses the session ?

Reply:
any sub-requests of a stateless (sub-)request are stateless also

Reply:
Therefore, I guess that `getCurrentRequest` will do the trick, am I right ?

Reply:
I'm not sure - any parent and explicitly stateless request should also trigger the exceptio

Reply:
Ok, got it, thanks